AccessHandle.md

```diff
@@ -0,0 +1,247 @@
+
+*   Allow tried and true technologies to be performantly used as part of web
+    applications e.g. using a port of your favorite storage library
+    within a website
+*   Distribute a Wasm module for WebSQL, allowing developers to us it across
+    browsers and opening the door to removing the unsupported API from Chrome
+*   Allow a music production website to operate on large amounts of media, by
+    relying on the new surface's performance and direct buffered access
+    to offload sound segments to disk instead of holding them in memory
+*   Provide a persistent [Emscripten](https://emscripten.org/) filesystem that
+    outperforms
+    [IDBFS](https://emscripten.org/docs/api_reference/Filesystem-API.html#filesystem-api-idbfs)
+    and has a simpler implementation
+
+## Non-goals
+
+This proposal is focused only on additions to the
+[Origin Private File System](https://wicg.github.io/file-system-access/#sandboxed-filesystem),
+and doesn't currently consider changes to the rest of File System Access API or
+how files in the host machine are accessed.
+
+## Proposed API
+
+### New data access surface
+
+```javascript
+//In all contexts
+const handle = await file.createAccessHandle();
+await handle.writable.getWriter().write(buffer);
+const reader = handle.readable.getReader({mode: "byob"});
+//Assumes seekable streams are available
+var {value: buffer, done} = await reader.read(buffer, {at: 1});
+
+//Only in a worker context
+const handle = file.createSyncAccessHandle();
+var writtenBytes = handle.write(buffer);
+var readBytes = handle.read(buffer {at: 1});
+```
+
+A new *createAccessHandle()* method would be added to *FileSystemFileHandle*.
+It  would return an *AccessHandle* that contains a
+[duplex stream](https://streams.spec.whatwg.org/#other-specs-duplex) and
+auxilliary methods. The readable/writable pair in the duplex stream would
+communicate with the same backing file, allowing the user to read unflushed
+contents. Another new method, *createSyncAccessHandle()*, would be only exposed
+on Worker threads. This method would offer a more buffer based surface for read
+and write. An IDL description of the new interface can be found in the
+[Appendix](#appendix).
+
+The reason for offering a Worker-only synchronous interface, is that consuming
+asynchronous APIs from Wasm has severe performance implications (more details
+[here](https://docs.google.com/document/d/1lsQhTsfcVIeOW80dr467Auud_VCeAUv2ZOkC63oSyKo)).
+We've opted for slightly different interfaces between the async and sync
+versions, because there currently isn’t support for synchronous streams. The
+amount of effort that would be required to spec them in a way that makes them
+compatible with asynchronous streams would be prohibitively high, and likely
+not worth it to support a single API.  Furthermore, we hope to discourage the
+use of the sync surface unless a developer is dealing with legacy Wasm ports,
+and therefore there is not much benefit to adding them.
```

Comment:
Could you rephrase that? It sounds like there is not much benefit in the sync interface at all

Reply:
Done.